Hide 'shared with me' grains that already show up as 'mine'.

dwrensha opened the pull request:
Fixes #1284

zarvox commented on shell/packages/sandstorm-ui-grainlist/grainlist-client.js:
```diff
@@ -123,12 +128,14 @@ SandstormGrainListPage.bulkActionButtons = function (showTrash) {
-          ownedGrainIds.forEach((grainId) => {
+          const grainIds = ownedGrainIds.concat(sharedGrainIds);
+
+          grainIds.forEach((grainId) => {
```
Is the idea here that we'll call both `deleteGrain` and `forgetGrain` on each grain to delete, regardless of whether we own the grain or have ApiTokens for it?  I find it a bit weird that we're sending requests over the wire that we know a priori will fail here.

kentonv replied:
I agree with this.

Maybe the `deleteGrain` method should be extended to automatically delete any owner-owned API tokens? Then you call `deleteGrain` if the grain is owned by the user and `forgetGrain` otherwise, but don't have to worry about the "both" case.

Or maybe these two methods should be merged into one with the server figuring out which one applies.

dwrensha commented:
> Maybe the deleteGrain method should be extended to automatically delete any owner-owned API tokens?

Done.

We could conceivably replace all instances of `forgetGrain(grainId, identityId)` with `deleteGrain(grainId)`, but I think it might prove useful to keep around the finer-grained API.

dwrensha commented:
Added another commit that tightens the semantics of `deleteGrain()` so that it could not actually be used as a substitute for `forgetGrain()`. Also cleaned up some cruft from `SandstormDb.deleteGrains()`. Rebased to avoid conflict with #2178.
